projects/avodata/AvodataSolverTool.java

```diff
@@ -0,0 +1,142 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Scanner;
+
+import org.apache.commons.math3.analysis.function.Abs;
+import org.apache.commons.math3.linear.Array2DRowRealMatrix;
+import org.apache.commons.math3.linear.DecompositionSolver;
+import org.apache.commons.math3.linear.MatrixUtils;
+import org.apache.commons.math3.linear.RealMatrix;
+import org.apache.commons.math3.linear.RealVector;
+import org.apache.commons.math3.linear.SingularValueDecomposition;
+import org.apache.commons.math3.util.CombinatoricsUtils;
+
+/** To build: bazel build //projects/avodata:ls-alg_tool To run: bazel-bin/projects/avodata/ls-alg_tool */
+/** This class reads matrix from csv file, ignores the parmeters and calculate 
+/** using least squares */
+public class AvodataSolverTool {
+
+  static final String fpath = "/projects/avodata/check.csv";
+
+  public static double factorial(int n) {
+    return CombinatoricsUtils.factorialDouble(n);
+  }
+
+  public static double nChooseK(int n, int k) {
+    return CombinatoricsUtils.binomialCoefficientDouble(n, k);
+  }
+
+  public static double[][] generateSyntheticA(int numOfProfessions) {
```

Comment:
I see sometimes you're returning `double[][]` and sometimes `RealMatrix`. It's better to be consistent.

Reply:
Ok, changed every method to return RealMatrix because it's the interface and more abstract.

Reply:
Cool